tests/mock_crust/churn.rs

```diff
@@ -90,6 +92,44 @@ fn did_receive_get_success(node: &TestNode,
     }
 }
 
+fn quorum_before(nodes: &[TestNode],
+                 src: Authority,
+                 dst: Authority,
+                 data_id: DataIdentifier,
+                 message_id: MessageId)
+                 -> usize {
+    let group = nodes.iter()
+        .filter(|n| n.routing_table().is_recipient(&src.to_destination()))
+        .collect_vec();
+
+    for node in &group {
+        let _ = node.inner.send_get_request(src, dst, data_id, message_id);
+    }
+
+    (group.len() * QUORUM - 1) / 100 + 1
+}
+
+fn quorum_after(nodes: &[TestNode],
+                src: Authority,
+                dst: Authority,
+                data_id: DataIdentifier,
+                message_id: MessageId)
+                -> (usize, usize) {
+    let group = nodes.iter()
+        .filter(|n| n.routing_table().is_recipient(&dst.to_destination()))
+        .collect_vec();
+    let quorum = (group.len() * QUORUM - 1) / 100 + 1;
+
+    let num_received = group.iter()
+                            .filter(|node| did_receive_get_request(node,
+                                                                   src,
+                                                                   dst,
+                                                                   data_id,
+                                                                   message_id))
+                            .count();
+    (quorum, num_received)
+}
```

Comment:
I think it would be better design if we had something like:

```rust
fn target_group(nodes: &[TestNode], target: Authority) -> Vec<&TestNode> {
    nodes.iter()
         .filter(|n| n.routing_table().is_recipient(&target.to_destination()))
         .collect_vec()
}

fn send_requests(...) {
    for node in &group {
        let _ = node.inner.send_get_request(src, dst, data_id, message_id);
    }
}

fn count_received(...) {
    group.iter()
         .filter(|node| did_receive_get_request(node,
                                                src,
                                                dst,
                                                data_id,
                                                message_id))
         .count()
}
```

and then used those in the tests along with `quorum()`. Wouldn't it be cleaner?